### PR TITLE
Made samples build independent from nonfree module

### PR DESCRIPTION
**Pull request by asmorkalov**

### CODE REVIEW

**Comment by asmorkalov** [comment]
@alalek please review.

**Comment by alalek** [comment]
@asmorkalov Please add fix like this into CMakeLists.txt files:
```
- ocv_include_modules(${OPENCV_C_SAMPLES_REQUIRED_DEPS})
+ ocv_include_modules(${OPENCV_C_SAMPLES_REQUIRED_DEPS} opencv_nonfree)
```

File: samples/cpp/tutorial_code/features2D/SURF_descriptor.cpp [hunk 1]
```diff
@@ -4,12 +4,23 @@
  * @author A. Huaman
  */
 
+#include "opencv2/opencv_modules.hpp"
 #include <stdio.h>
-#include <iostream>
```
Comment (alalek) [inline samples/cpp/tutorial_code/features2D/SURF_descriptor.cpp]: `#include <iostream>` is lost

**Comment by alalek** [comment]
:+1:

**Pull request merged**